tools/wptrunner/wptrunner/testloader.py

```diff
@@ -430,8 +430,11 @@ def make_queue(cls, tests_by_type, **kwargs):
 
         for item in groups:
             test_queue.put(item)
-        cls.add_sentinal(test_queue, kwargs["processes"])
-        return test_queue
+
+        # Do not launch more workers than the number of groups
+        processes = max(1, min(len(groups), kwargs["processes"]))
```

Comment:
Given we repeat this three times, it might be worthwhile making a function like
```
def process_count(requested_processes, num_test_groups):
    """Get the number of processes to use.

    This must always be at least one, but otherwise not more than the number of test groups"""
    return max(1, min(requested_processes, num_test_groups))

Reply:
Updated, thanks for the suggestion.